### PR TITLE
Issue 34488:Remove Abstract Component

### PR DESCRIPTION
**Pull request by ricardojaferreira**
TransportAction and BaseRestHandler now no longer extends AbstractComponent. The AbstractComponent no longer has usages so it was deleted. 

Relates to #34488

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java [hunk 1]
```diff
@@ -18,24 +18,25 @@
...
 
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> extends AbstractComponent {
+public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> {
 
     protected final String actionName;
     private final ActionFilter[] filters;
     protected final TaskManager taskManager;
+    private static final Logger logger = LogManager.getLogger(TransportAction.class);
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: I'm not sure that we want every action extending this to use the same logger. Also the visivility won't work, this leaves a lot of compile errors for me in "server".

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java [hunk 2]
```diff
@@ -51,12 +53,13 @@
...
  * {@link ActionPlugin#getRestHeaders()}.
  */
-public abstract class BaseRestHandler extends AbstractComponent implements RestHandler {
+public abstract class BaseRestHandler implements RestHandler {
 
     public static final Setting<Boolean> MULTI_ALLOW_EXPLICIT_INDEX =
         Setting.boolSetting("rest.action.multi.allow_explicit_index", true, Property.NodeScope);
 
     private final LongAdder usageCount = new LongAdder();
+    private static final Logger logger = LogManager.getLogger(BaseRestHandler.class);
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: This logger appears to be unused.

**Review by cbuescher — CHANGES_REQUESTED** [review]
@ricardojaferreira I left two comments, also please note that this change in its current form doesn't compile and there are PRs that are still not merged that are necessary to go in before we are actually able to remove this class entirely (e.g. https://github.com/elastic/elasticsearch/pull/35394)

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java (hunk 2 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: I was thinking of having something like:
```
@Deprecated
protected static Logger logger = LogMangaer.getLogger(getClass());
```

in 6.x and master and then removing it entirely from master in a follow up change.

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java (hunk 1 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: I think we should do the same thing here as we do for `BaseRestHandler`.

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java [hunk 3]
```diff
@@ -18,24 +18,26 @@
...
 
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> extends AbstractComponent {
+public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> {
 
     protected final String actionName;
     private final ActionFilter[] filters;
     protected final TaskManager taskManager;
+    @Deprecated
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: Why is this logger deprecated?

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java [hunk 4]
```diff
@@ -51,12 +53,14 @@
...
  */
-public abstract class BaseRestHandler extends AbstractComponent implements RestHandler {
+public abstract class BaseRestHandler implements RestHandler {
 
     public static final Setting<Boolean> MULTI_ALLOW_EXPLICIT_INDEX =
         Setting.boolSetting("rest.action.multi.allow_explicit_index", true, Property.NodeScope);
 
     private final LongAdder usageCount = new LongAdder();
+    @Deprecated
+    protected static Logger logger = LogManager.getLogger(BaseRestHandler.class);
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: why is this logger deprecated?

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java [hunk 5]
```diff
@@ -18,24 +18,26 @@
  */
 
 package org.elasticsearch.action.support;
-
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.message.ParameterizedMessage;
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: that seems to be unused, do we need that? (won't pass the build)

**Comment by pgomulka** [comment]
@ricardojaferreira great effort, thank you. I left some comments. Make sure the build passes to you locally. Definitely ./gradlew assemble precommit. Consider running ./gradlew check to see all tests passing

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java [hunk 6]
```diff
@@ -18,24 +18,26 @@
...
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> extends AbstractComponent {
+public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> {
 
     protected final String actionName;
     private final ActionFilter[] filters;
     protected final TaskManager taskManager;
+    @Deprecated
+    protected static Logger logger = LogManager.getLogger(TransportAction.class);
```
Comment (pgomulka) [inline server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: should this be for getClass() instead of TransportAction.class?

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java (hunk 4 above)
Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: should this be for getClass() instead of BaseRestHandler.class?

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java (hunk 6 above)
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: Using "LogManager.getLogger(TransportAction.class)" would still mean all inheriting classes using this would share the same logger. I don't think this is what we want. Something like "LogManager.getLogger(getClass())" might work for now. Also: why did you add the "@Deprecated" annotation? The loggers themselves are not ment to be deprecated as far as I understand #34488 .

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java (hunk 4 above)
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: same here as above.

**Review by pgomulka — COMMENTED** [review]
@ricardojaferreira great effort, thank you. I left some comments. Make sure the build passes to you locally. Definitely ./gradlew assemble precommit. Consider running ./gradlew check to see all tests passing

**Review by cbuescher — CHANGES_REQUESTED** [review]
Hi @ricardojaferreira, as far as I see your last commit only changed the visibility of the loggers in TransportAction and BaseRestHandler but their would still share the same logger with all inheriting classes. I left a comment along those lines. I might not be totally up to date with the current state of the final works on #34488 so I ask @nik9000 and @pgomulka to also take a look because they will know the remaining other work that needs to get done around it.

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java (hunk 3 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: I think that subclasses of `TransportAction` should declare their own logger. But I think it'd be kinder that that were only a hard requirement for 7.0 so making a deprecated logger here is nice, so long as we follow up to zap it.

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java (hunk 4 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: Yeah, `getClass()` I think.

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java (hunk 6 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: I like deprecating them because I think everything that wants a logger should declare one. That way we don't have a zillion loggers we don't use. I do think that if you deprecate something you should add javadoc explaining what to do instead. Something like:
```
/**
 * @deprecated declare your own logger.
 */
@Deprecated
protected static Logger logger = LogManager.getLogger(TransportAction.class);
```

File: server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java (hunk 4 above)
Comment (ricardojaferreira) [reply server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: Hi, Thanks for the comments. I am still understanding how the project works. Anyway, to use `getClass()` the logger cannot be static. Is that ok? (Or am I seeing something wrong here)

Comment (ricardojaferreira) [reply server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java]: Hi, Thanks for the comments. I am still understanding how the project works. Anyway, to use `getClass()` the logger cannot be static. Is that ok? (Or am I seeing something wrong here)

File: server/src/main/java/org/elasticsearch/action/support/TransportAction.java (hunk 6 above)
Comment (ricardojaferreira) [reply server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: Same has above, use getClass with static?

Comment (ricardojaferreira) [reply server/src/main/java/org/elasticsearch/action/support/TransportAction.java]: I will add the comment on deprecated loggers.

**Comment by pgomulka** [comment]
@elasticmachine test this please

**Review by cbuescher — APPROVED** [review]
Thanks for the update, LGTM

**Comment by pgomulka** [comment]
Ok to test

**Comment by pgomulka** [comment]
Ok to test

**Comment by pgomulka** [comment]
@ricardojaferreira could you please upstream master into this PR? 
Or do you mind if I do it instead of you on your master?

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Review by pgomulka — APPROVED** [review]
LGTM. Build passes

**Comment by pgomulka** [comment]
@ricardojaferreira the PR now passed the build and I have merged your change. Thank you once again for you contribution!
